test/performance/select_perf_test.lua

```diff
@@ -0,0 +1,177 @@
+                servers = {
+                    { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                },
+            },
+            {
+                uuid = helpers.uuid('b'),
+                alias = 's-1',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('b', 1), alias = 's1-master' },
+                    { instance_uuid = helpers.uuid('b', 2), alias = 's1-replica' },
+                },
+            },
+            {
+                uuid = helpers.uuid('c'),
+                alias = 's-2',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('c', 1), alias = 's2-master' },
+                    { instance_uuid = helpers.uuid('c', 2), alias = 's2-replica' },
+                },
+            },
+            {
+                uuid = helpers.uuid('d'),
+                alias = 's-2',
+                roles = { 'customers-storage', 'crud-storage' },
+                servers = {
+                    { instance_uuid = helpers.uuid('d', 1), alias = 's3-master' },
+                    { instance_uuid = helpers.uuid('d', 2), alias = 's3-replica' },
+                },
+            }
+        },
+    })
+    g.cluster:start()
+end
+
+g.after_all = function()
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+g.before_each(function() end)
+
+local function insert_customers(conn, id, count, timeout, report)
+    local customer = {id, box.NULL, 'David', 'Smith', 33, 'Los Angeles'}
+    local start = fiber.clock()
+
+    while (fiber.clock() - start) < timeout do
+        local ok, res, err = pcall(conn.call, conn, [[package.loaded.crud.insert]], {'customers', customer})
+        if not ok then
+            log.error('Insert error: %s', res)
+            table.insert(report.errors, res)
+        elseif err ~= nil then
+            errors.wrap(err)
+            log.error('Insert error: %s', err)
+            table.insert(report.errors, err)
+        else
+            report.count = report.count + 1
+        end
+        customer[1] = customer[1] + count
+    end
+end
+
+local function select_customers(conn, id, timeout, report)
+    local start = fiber.clock()
+    local ok, err = pcall(function()
+        while (fiber.clock() - start) < timeout do
+            local _, err = conn:call([[package.loaded.crud.select]], {'customers', {{'>', 'id', id}}, {first = 10}})
+            if err ~= nil then
+                errors.wrap(err)
+                log.error(err)
+                table.insert(report.errors, err)
+            else
+                report.count = report.count + 1
+            end
+        end
+    end)
+    if not ok then
+        table.insert(report.errors, err)
+        log.error(err)
+    end
+end
+
+g.test_insert = function()
+    local timeout = 30
+    local fiber_count = 600
```

Comment:
These parameters are always static. Do we need to calculate these parameters dynamically depending on RAM (for example)?

Reply:
I don't know how we could calculate it. We don't consume huge amount of RAM and all transactions are processed in single thread.